tests/iam/permhandler/permhandler_test.cpp

```diff
@@ -0,0 +1,211 @@
+ * Tests
+ **********************************************************************************************************************/
+
+TEST_F(PermHandlerTest, RegisterInstanceSucceeds)
+{
+    InstanceIdent                                instanceIdent;
+    StaticArray<FunctionalServicePermissions, 1> perms;
+    const auto                                   err = mPermHandler.RegisterInstance(instanceIdent, perms);
+
+    EXPECT_TRUE(err.mError.IsNone()) << err.mError.Message();
+
+    ASSERT_FALSE(err.mValue.IsEmpty());
+}
+
+TEST_F(PermHandlerTest, RegisterInstanceReturnsSecretFromCache)
+{
+    InstanceIdent                                instanceIdent;
+    StaticArray<FunctionalServicePermissions, 1> perms;
+
+    const auto lhsErr = mPermHandler.RegisterInstance(instanceIdent, perms);
+
+    EXPECT_TRUE(lhsErr.mError.IsNone()) << lhsErr.mError.Message();
+    ASSERT_FALSE(lhsErr.mValue.IsEmpty());
+
+    const auto rhsErr = mPermHandler.RegisterInstance(instanceIdent, perms);
+    EXPECT_EQ(lhsErr.mError, rhsErr.mError);
+    EXPECT_EQ(lhsErr.mValue, rhsErr.mValue);
+}
+
+TEST_F(PermHandlerTest, RegisterInstanceReachedMaxSize)
+{
+    for (size_t i {0}; i < cMaxNumInstances; ++i) {
+        const InstanceIdent                          instanceIdent {"", "", i};
+        StaticArray<FunctionalServicePermissions, 1> perms;
+
+        const auto err = mPermHandler.RegisterInstance(instanceIdent, perms);
+        ASSERT_TRUE(err.mError.IsNone()) << err.mError.Message();
+        ASSERT_FALSE(err.mValue.IsEmpty());
+    }
+
+    const InstanceIdent                          instanceIdent {"", "", cMaxNumInstances};
+    StaticArray<FunctionalServicePermissions, 1> perms;
+
+    const auto err = mPermHandler.RegisterInstance(instanceIdent, perms);
+    ASSERT_TRUE(err.mError.Is(ErrorEnum::eNoMemory)) << err.mError.Message();
+    ASSERT_TRUE(err.mValue.IsEmpty());
+}
+
+TEST_F(PermHandlerTest, GetPermissionsNotFound)
+{
+    InstanceIdent                instanceIdent;
+    StaticArray<PermKeyValue, 1> perms;
+
+    const auto err = mPermHandler.GetPermissions(instanceIdent, perms, "uknownSecretUUID", "unknownServerId");
+    ASSERT_TRUE(err.Is(ErrorEnum::eNotFound)) << err.Message();
+}
+
+TEST_F(PermHandlerTest, GetPermissionsNoMemoryForPerms)
+{
+    const InstanceIdent instanceIdent1 {"serviceID1", "subjectID1", 1};
+
+    FunctionalServicePermissions testServicePermissions;
+    testServicePermissions.mName = "testService";
+    testServicePermissions.mPermissions.PushBack({"key1", "value1"});
+    testServicePermissions.mPermissions.PushBack({"key2", "value2"});
+
+    StaticArray<FunctionalServicePermissions, 2> funcServerPermissions;
+    funcServerPermissions.PushBack(testServicePermissions);
+
+    const auto errRegister = mPermHandler.RegisterInstance(instanceIdent1, funcServerPermissions);
+    ASSERT_TRUE(errRegister.mError.IsNone()) << errRegister.mError.Message();
+
+    const auto secret = errRegister.mValue;
```

Comment:
Why intermediate var is required?

Reply:
Refactored. Currently this UT has 2 secret variables (for 2 different instances) that are used in test